overlord/ifacestate/helpers.go

```diff
@@ -191,6 +194,45 @@ func (m *InterfaceManager) renameCorePlugConnection() error {
 	return nil
 }
 
+// removeStaleConnections removes stale connections left by some older versions of snapd.
+// Connection is considered stale if the snap on either end of the connection doesn't exist anymore.
+// XXX: this code should eventually go away.
+func (m *InterfaceManager) removeStaleConnections() error {
+	conns, err := getConns(m.state)
+	if err != nil {
+		return err
+	}
+	var staleConns int
+	for id := range conns {
+		connRef, err := interfaces.ParseConnRef(id)
+		if err != nil {
+			return err
+		}
+		var snapst snapstate.SnapState
+		if err := snapstate.Get(m.state, connRef.PlugRef.Snap, &snapst); err != nil {
+			if err != state.ErrNoState {
+				return err
+			}
+			delete(conns, id)
+			staleConns++
+			continue
+		}
+		if err := snapstate.Get(m.state, connRef.SlotRef.Snap, &snapst); err != nil {
+			if err != state.ErrNoState {
+				return err
+			}
+			delete(conns, id)
+			staleConns++
+			continue
+		}
+	}
+	if staleConns > 0 {
+		setConns(m.state, conns)
+		logger.Noticef("remove %d stale connections", staleConns)
```

Comment:
Nitpick: use past tense please, `removed`. Alternatively we could log the actually deleted connections because after seeing this message I will only wonder, "gee which connections were removed?"